content.js

```diff
@@ -529,53 +530,53 @@ function reskipSponsorTime(UUID) {
-    startSponsorButton.addEventListener("click", startSponsorClicked);
-
-    let startSponsorImage = document.createElement("img");
-    startSponsorImage.id = "startSponsorImage";
-    startSponsorImage.className = "playerButtonImage";
-    startSponsorImage.src = chrome.extension.getURL("icons/PlayerStartIconSponsorBlocker256px.png");
-
-    //add the image to the button
-    startSponsorButton.appendChild(startSponsorImage);
-
-    let controls = document.getElementsByClassName("ytp-right-controls");
-    let referenceNode = controls[controls.length - 1];
-
-    if (referenceNode == undefined) {
-        //page not loaded yet
-        setTimeout(addPlayerControlsButton, 100);
-        return;
-    }
-
-    referenceNode.prepend(startSponsorButton);
-}
-
 function removePlayerControlsButton() {
-    document.getElementById("startSponsorButton").style.display = "none";
+	if (!sponsorVideoID) return;
+    document.getElementById("changeStartSponsor").style.display = "none";
     document.getElementById("submitButton").style.display = "none";
 }
 
+function createButton(baseid, title, callback, imageName) {
+  if (document.getElementById(baseid+"Button") != null) return;
+  // Button HTML
+  let newButton = document.createElement("button");
+  newButton.id = baseid+"Button";
+  newButton.className = "ytp-button playerButton";
+  newButton.setAttribute("title", chrome.i18n.getMessage(baseid));
+  newButton.addEventListener("click", callback);
+  // Image HTML
+  let newButtonImage = document.createElement("img");
+  newButtonImage.id = baseid+"Image";
+  newButtonImage.className = "playerButtonImage";
+  newButtonImage.src = chrome.extension.getURL("icons/"+imageName);
+  // Append image to button
+  newButton.appendChild(newButtonImage);
+  // Add the button to player
+  controls.prepend(newButton);
+}
+
+function getControls() {
+	  let controls = document.getElementsByClassName("ytp-right-controls");
+	  return (!controls || controls.length === 0) ? false : controls[controls.length - 1]
+};
+
+//adds the player controls buttons
+function addButtons() {
+  wait(getControls).then(result => {
+	  controls = result; // Global
+	  
+	  // Add button if does not already exist in html
+	  createButton("startSponsor", "sponsorStart", startSponsorClicked, "PlayerStartIconSponsorBlocker256px.png");	  
+      createButton("info", "openPopup", openInfoMenu, "PlayerInfoIconSponsorBlocker256px.png")
+	  createButton("delete", "clearTimes", clearSponsorTimes, "PlayerDeleteIconSponsorBlocker256px.png");
+      createButton("submit", "SubmitTimes", submitSponsorTimes, "PlayerUploadIconSponsorBlocker256px.png");
+  });
+}
 //adds or removes the player controls button to what it should be
 function updateVisibilityOfPlayerControlsButton() {
     //not on a proper video yet
-    if (!getYouTubeVideoID(document.URL)) return;
-
-    addPlayerControlsButton();
-    addInfoButton();
-    addDeleteButton();
-    addSubmitButton();
+    if (!sponsorVideoID) return;
+	
```

Comment:
You should still try adding the buttons here as well.

Reply:
Seems like a bad idea running createButton even after its added

Reply:
They should have checks in the function

Reply:
How will this change improve the extension?
The buttons should never get removed

Reply:
This is just in case the other function doesn't add them in time.

Reply:
The change visibility function is used a lot I think its more likely to forget a check